py/desispec/desi_proc_dashboard.py

```diff
@@ -0,0 +1,383 @@
+import argparse
+import os,glob
+import fitsio
+import astropy.io.fits as pyfits
+from astropy.io import fits
+import subprocess
+import pandas as pd
+import time,datetime
+import numpy as np
+import psycopg2
+import hashlib
+import pdb
+import psutil
+from os import listdir
+
+class DESI_PROC_DASHBOARD(object):
+    """ Code to generate the statistic of desi_pipe production status   
+    Usage:
+    Cron job script:
+    */30 * * * * /global/common/software/desi/cori/desiconda/20190804-1.3.0-spec/conda/bin/python3 /global/project/projectdirs/desi/users/zhangkai/desi_code/desi_proc_dashboard.py -n all -nn 60 --prod_dir /global/cscratch1/sd/zhangkai/desi/ --output_dir /global/project/projectdirs/desi/www/users/zhangkai/desi_proc_dashboard/ --output_url https://portal.nersc.gov/project/desi/users/zhangkai/desi_proc_dashboard/  >/global/project/projectdirs/desi/users/zhangkai/desi_proc_dashboard.log 2>/global/project/projectdirs/desi/users/zhangkai/desi_proc_dashboard.err &
+
+    """
+
+    def __init__(self):
+        ############
+        ## Input ###
+        ############
+        parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+        parser = self._init_parser(parser)
+        args = parser.parse_args()
+        if not os.getenv('DESI_SPECTRO_REDUX'):
+            os.environ['DESI_SPECTRO_REDUX']='/global/project/projectdirs/desi/spectro/redux/'
+            os.environ['DESI_SPECTRO_DATA']='/global/project/projectdirs/desi/spectro/data/'
+        if args.nights=='all':
+            nights=listdir(os.getenv('DESI_SPECTRO_REDUX')+'/daily/exposures/')
```

Comment:
Throughout the code, use `args.prod_dir` with that having a default of `desispec.io.specprod_root()` (which currently just returns `$DESI_SPECTRO_REDUX/$SPECPROD), instead of hardcoding "daily" as the only production name that this can be used on.  Similarly use `desispec.io.rawdata_root()` to get the location of the raw data.

Elsewhere it looks like you are using `args.prod_dir` as the equivalent of `$DESI_ROOT`, i.e. `/global/project/projectdirs/desi` -> `/global/cfs/cdirs/desi`.  Let's move the meaning of prod_dir a few levels deeper (i.e. all the way into the spectro/redux/daily/ directory, without hardcoding that name).

Reply:
Realize that prod_dir is not used in fact, thus remove it.

Reply:
args.prod_dir is not pointing to desispec.io.specprod_root()